specs/badge.md

```diff
@@ -0,0 +1,83 @@
+# fast-badge
+
+## Overview
+
+*Badge* component is used to highlight an item and attract attention or flag status, such as “New”, “Sale”, or other short phrases or a number to represent unread messages, updates available, etc..
+
+### Use Cases
+
+Typical use cases include, but are not limited to, denoting a sale or new item, flagging an item as part of a category or representing a value of unread messages.
+  
+### Features
+
+A badge should allow the following properties:
+- `fill`, a string that passes the color value for the backplate. Defaults to "transparent".
+- `circular`, if circular styling is desired.
+- `size`, the size the badge is rendered.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/badge)
+- [Material UI](https://material-ui.com/components/badges/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/badges/)
+- [Ant Design Badge](https://ant.design/components/badge/)
+- [Ant Design Lozenge](https://atlaskit.atlassian.com/packages/core/lozenge)
+- [Atlassian](https://atlaskit.atlassian.com/packages/core/badge)
+
+---
+
+### API
+
+*Component name:*
+- `fast-badge`
+
+*Attributes:*
+- `fill`: string
+- `circular`: boolean
+- `size`: enum
+  - small
+  - medium
+  - large
+
+*Slots:*
+- `default`
+
+### Anatomy and Appearance
+
+*Template:*
+```
+<slot></slot>
+```
+
+## Implementation
+
+```
+fast-badge {
+   --fill-primary: rgba(255, 0, 0, 1);
+   --fill-secondary: #00FF00;
+   --fill-transparent: transparent;
```

Comment:
We don't use primary anywhere as of now - what would this map to? Typically we have avoided "primary" and "secondary" (see button). I would expect some alignment across the system; is this a recommendation for a system wide change? If so, I think it's worthy of its own thread.

Reply:
These are app author defined variables, they can be named whatever they choose.

Reply:
Can you elaborate on how that works from a technical standpoint? For instance, if I want three states with my own semantics, how does that get set on our end? 

In addition to the above, if these are defined by the app author, can we make sure that's clear in the spec? I was reading it as though those are set by us, as they appear to have the values being defined and not both the property and the values.

Reply:
In the template:
```
<div style="background-color: var(--badge-fill-${x => x.fill})">
   ...
</div>
```

Implementation
```
<style>
  fast-badge {
    --badge-fill-foo: red;
  }
</style>

<fast-badge fill="foo"/>

```
I can make this clearer in the spec.